source/clear-linux/guides/network/firewalls.rst

```diff
@@ -0,0 +1,198 @@
+attributes like IP address, port, payload and more. 
+
+The `Netfilter <https://netfilter.org/>`_ framework in the Linux kernel
+performs packet filtering and provides the means for implementing a software
+firewall in Linux. |CL-ATTR| has a couple different firewall front-end options
+for managing the Linux firewall.
+
+
+.. contents:: :local:
+    :depth: 2
+   
+
+Default ruleset
+***************
+
+|CL| does not impose a firewall policy out of the box. All traffic is allowed
+inbound and all traffic is allowed outbound. However, `tallow`_ is installed
+by default and may dynamically create rules temporarily restricting SSH
+access.
+
+.. warning::
+
+   Changing firewall configuration can cause abrupt network disconnection, if
+   misconfigured. If this happens on a remote host, local recovery may be
+   required.
+   
+   Be sure to test your firewall configuration before committing it permanently.
+
+
+iptables 
+********
+
+:command:`iptables` is a well-known user-space administration tool for
+configuring IPv4 Linux firewall rules. :command:`ip6tables` is a
+complimentary tool for configuring IPv6 Linux firewall rules. 
+
+Below is information on using :command:`iptables` on |CL|: 
+
+#. Make sure the *iptables* bundle is installed
+
+   .. code:: bash
+
+      sudo swupd bundle-add iptables
+
+
+#. Define new iptables rules/chains for the running configuration using the
+   :command:`iptables` command. See :command:`man iptables` for iptables
+   concepts and configuration options.
+
+   Below is a common restrictive firewall configuration which denies all
+   incoming connections, unless the connection was initiated by the host. 
+
+   .. code:: bash
+
+      #Delete all existing rules 
+      sudo iptables -F
+
+      # Set default chain policies 
+      sudo iptables -P INPUT DROP
+      sudo iptables -P FORWARD DROP
+      sudo iptables -P OUTPUT ACCEPT
+
+      # Accept on localhost loopback device
+      sudo iptables -A INPUT -i lo -j ACCEPT
+      sudo iptables -A OUTPUT -o lo -j ACCEPT
+
+      # Allow established sessions to receive traffic
+      sudo iptables -A INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT
+
+
+#. Test the running firewall configuration to make sure it behaving as
+   you expect. 
+
+#. Run the *iptables-save* service to make the running configuration
+   persistent. This will perform a one-time save of the running configuration
+   to :file:`/etc/iptables.rules` :
+
+   .. code:: bash
+
+      sudo systemctl start iptables-save
+
+
+
+
+#. Enable the *iptables-resolve* service the iptables rules to be
+   automatically applied at boot from the :file:`/etc/iptables.rules` file:
+
+   .. code :: bash
+
+      sudo systemctl enable iptables-restore.service
+
+
+ipset
+=====
+
+`ipset <http://ipset.netfilter.org/>`_ is a framework in the Linux kernel for
+storing and efficiently indexing combinations of IP addresses, networks,
+(TCP/UDP) port numbers, MAC addresses, and interface names.
```

Comment:
"By itself, `ipset` does not enforce network traffic rules. It acts as a simple but efficient table that allows `netfilter` rules to match network packets against a large set of entries. This makes writing network policy rules simpler, but also a lot more efficient."